lib/cocoapods-downloader/http.rb

```diff
@@ -3,20 +3,30 @@
 module Pod
   module Downloader
     class Http < RemoteFile
+      USER_AGENT_HEADER = 'User-Agent'.freeze
+
       private
 
       executable :curl
 
       def download_file(full_filename)
         parameters = ['-f', '-L', '-o', full_filename, url, '--create-dirs', '--netrc-optional', '--retry', '2']
-
+        parameters << user_agent_argument if headers.nil? || headers.none? { |header| header.include?(USER_AGENT_HEADER) }
```

Comment:
I think header names are case-insensitive, so should we do a case-insensitive comparison here?